dnfdaemon-client/commands/command.cpp

```diff
@@ -39,11 +42,39 @@ void TransactionCommand::run_transaction(Context & ctx) {
     // resolve the transaction
     options["allow_erasing"] = ctx.allow_erasing.get_value();
     std::vector<dnfdaemon::DbusTransactionItem> transaction;
+    dnfdaemon::KeyValueMap dbus_goal_resolve_results;
     ctx.session_proxy->callMethod("resolve")
         .onInterface(dnfdaemon::INTERFACE_GOAL)
         .withTimeout(static_cast<uint64_t>(-1))
         .withArguments(options)
-        .storeResultsTo(transaction);
+        .storeResultsTo(transaction, dbus_goal_resolve_results);
+
+    // TODO (nsella): handle localization
+    if (!key_value_map_get<std::string>(dbus_goal_resolve_results, "transaction_solver_problems").empty()) {
+        std::cout << key_value_map_get<std::string>(dbus_goal_resolve_results, "transaction_solver_problems") << std::endl;
+    }
+    dnfdaemon::KeyValueMapList goal_resolve_log_list = key_value_map_get<dnfdaemon::KeyValueMapList>(dbus_goal_resolve_results, "goal_problems");
+
+    for (const auto & e : goal_resolve_log_list) {
+        libdnf::GoalAction action = static_cast<libdnf::GoalAction>(key_value_map_get<uint32_t>(e, "action"));
+        libdnf::GoalProblem problem = static_cast<libdnf::GoalProblem>(key_value_map_get<uint32_t>(e, "problem"));
+        libdnf::GoalJobSettings job_settings;
+        job_settings.to_repo_ids = key_value_map_get<dnfdaemon::KeyValueMap>(e, "goal_job_settings").at("to_repo_ids");
+        std::string report = key_value_map_get<std::string>(e, "report");
+        std::vector<std::string> report_list = key_value_map_get<std::vector<std::string>>(e, "report_list");
+        std::set<std::string> report_set { report_list.begin(), report_list.end() };
+
+        std::string format_log = libdnf::base::Transaction::format_resolve_log(
+                action,
+                problem,
+                job_settings,
+                report,
+                report_set);
+        if (!format_log.empty()) {
+            std::cout << format_log << std::endl;
+        }
+    }
+    return;
```

Comment:
This `return` should be conditionalized - only in case the `resolve()` method returns something other than `NO_PROBLEM`.